src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/SelectEmployerViewModelMapper.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.ProviderCommitments.Web.Models;
+using SFA.DAS.ProviderCommitments.Web.Requests;
+using SFA.DAS.ProviderRelationships.Api.Client;
+using SFA.DAS.ProviderRelationships.Types.Dtos;
+using SFA.DAS.ProviderRelationships.Types.Models;
+using SFA.DAS.ProviderUrlHelper;
+
+namespace SFA.DAS.ProviderCommitments.Web.Mappers
+{
+    public class SelectEmployerViewModelMapper : IMapper<SelectEmployerRequest, SelectEmployerViewModel>
+    {
+        private readonly IProviderRelationshipsApiClient _providerRelationshipsApiClient;
+        private readonly ILinkGenerator _linkGenerator;
+
+        public SelectEmployerViewModelMapper(IProviderRelationshipsApiClient providerRelationshipsApiClient, ILinkGenerator linkGenerator)
+        {
+            _providerRelationshipsApiClient = providerRelationshipsApiClient;
+            _linkGenerator = linkGenerator;
+        }
+
+        public async Task<SelectEmployerViewModel> Map(SelectEmployerRequest source)
+        {
+            return new SelectEmployerViewModel
+            {
+                AccountProviderLegalEntities = (await GetLegalEntitiesWithCreatePermission(source.ProviderId)).Select(x=>new AccountProviderLegalEntityViewModel
+                {
+                    EmployerAccountLegalEntityName = x.AccountLegalEntityName,
+                    EmployerAccountLegalEntityPublicHashedId = x.AccountLegalEntityPublicHashedId,
+                    EmployerAccountName = x.AccountName,
+                    EmployerAccountPublicHashedId = x.AccountPublicHashedId,
+                }).ToList(),
+                BackLink = _linkGenerator.ProviderApprenticeshipServiceLink("account")
```

Comment:
We've tended to make links and routing etc a view concern, rather than inject the link generator into the vm mappers. Can we move this out for consistency? It's a simple one anyway, no logic to it etc.